### ISSUE TITLE
Are there up-to-date instructions on how to start using Sparkle 2.x?

### ISSUE THREAD

**Issue by SamusAranX** [sparkle-project/Sparkle#1677]
I have never used Sparkle in any real capacity before. I played with it 5-6 years ago but never got anywhere.

Now I'm trying to integrate Sparkle 2.x into a sandboxed app of mine that's written in Swift. The instructions I could find here: https://github.com/sparkle-project/Sparkle/blob/2.x/INSTALL.markdown or here: https://christiantietze.de/posts/2019/06/sparkle-xpc-setup/ get me as far as "copying Sparkle.framework into your project's Frameworks directory" and that's it. There doesn't seem to be any documentation beyond that.
What do I need to do in order to get Sparkle to fully work with my app?

**Comment by SamusAranX** [comment]
I was told to check out NetNewsWire since they're using Sparkle as well and that helped me with the initial problem of integrating it into my project.

The problem I'm now facing is that Sparkle is not recognizing the public key I put into my Info.plist.
![image](https://user-images.githubusercontent.com/676069/99586466-1536da80-29e8-11eb-9b98-45660340fd73.png)
As per https://sparkle-project.org/documentation/customization/, I pasted the public key into my Info.plist (as a multiline string, because that's what it is), yet Sparkle keeps going `[Sparkle] The provided EdDSA key could not be decoded.`.

What form does my public key have to be in for Sparkle to be accepted? Cursory searches for `SUPublicEDKey` on GitHub show me that people are including much shorter forms of the public key in their Info.plist, but it's not clear to me how to turn my 20-line key into this short one, and whether this even still applies to 2.x. The upgrading manual for 1.x versions states
> Run bin/generate_keys. It will generate a new EdDSA keypair and print a public key.

but for me, running `generate_keys` doesn't:

```
$ submodules/Sparkle/bin/generate_keys
Generating DSA parameters, 2047 bit long prime
This could take some time
........+.....+...+...+..........+.+........+++++++++++++++++++++++++++++++++++++++++++++++++++*
.........+..+....................................+.....+..+..........+.................+.......+................+..+.+.......+...+......+..+.....+.+.+....+..+.+.....+.+........+....+....+...................+..+............+...........+............................+........+.+.+++++++++++++++++++++++++++++++++++++++++++++++++++*
Generating DSA key, 2048 bits
read DSA key
writing DSA key

Generated two files:
dsa_priv.pem: your private key. Keep it secret and don't share it!
dsa_pub.pem: public counterpart to include in the app bundle.

BACK UP YOUR PRIVATE KEY AND KEEP IT SAFE!
If you lose it, your users will be unable to upgrade!
```

I would just check in the NetNewsWire project, but they seem to have expunged all mentions of their Sparkle public key from it.

**Comment by SamusAranX** [comment]
By total chance, I noticed the output I pasted above didn't match the actual generate_keys code, so I compiled a fresh copy and that did the trick.

**Comment by zorgiepoo** [comment]
`INSTALL.markdown` was removed in favor of the documentation on https://sparkle-project.org

The signing scripts in the source repo in ./bin/ in 2.x were out of date (DSA is deprecated). That is being fixed in #1722. You would have however gotten the right generate_keys if you built a `Distribution` or done `make release`.

**Comment by zorgiepoo** [comment]
INSTALL document and documentation has been updated since then.
